#574: Adding webint example as service API (#575)
rticommunity/rticonnextdds-examples: examples/web_integration_service/library_api/c++11/README.md

# Example Code: Web Integration Service Library API in C++11

## Building the Example :wrench:

To build this example, first run CMake to generate the corresponding build
files. We recommend you use a separate directory to store all the generated
files (e.g., ./build).

```sh
mkdir build
cd build
cmake ..
```

Once you have run CMake, you will find a number of new files in your build
directory (the list of generated files will depend on the specific CMake
Generator). To build the example, run CMake as follows:

```sh
cmake --build .
```

**Note**: If you are using a multi-configuration generator, such as Visual
Studio solutions, you can specify the configuration mode to build as follows:

```sh
cmake --build . --config Release|Debug
```

Alternatively, you can use directly the generated infrastructure (e.g.,
Makefiles or Visual Studio Solutions) to build the example. If you generated
Makefiles in the configuration process, run make to build the example. Likewise,
if you generated a Visual Studio solution, open the solution and follow the
regular build process.

## Running the Example

Once the example is built it will generate the `WebIntegrationServiceLibraryAPICXX11`

> executable. This executable takes two arguments that are similar to the Web

Reviewer: ```suggestion
executable. This executable takes three arguments that are similar to the Web
```

Integration Service executable shipped with the Connext installation:

- `-cfgFile`: Specifies the QoS configuration file name

> - `-cfgName`: Specified the top level configuration name

Reviewer: ```suggestion
- `-cfgName`: Specifies the top level configuration name
```

> - `-documentRoot`: Specified the path to be served by the web server

Reviewer: ```suggestion
- `-documentRoot`: Specifies the path to be served by the web server
```

You can run the generated executable from the `build` folder as follows:

```sh

> ./WebIntegrationServiceLibraryAPICXX11 -cfgFile ../WebIntegrationServiceConfig.xml

Reviewer: Add -cfgName and -documentRoot

```

**Note:** The file `WebIntegrationServiceConfig.xml` has only one configuration called
`LibraryAPIDemo` that starts the Web Integration Service instance on HTTP port `8080`.

> This port can be changed by adding the corresponding option to tte serviceProperty

Reviewer: ```suggestion
This port can be changed by adding the corresponding option to `service_property`
```
